zeppelin-server/src/test/java/org/apache/zeppelin/rest/ZeppelinSparkClusterTest.java

```diff
@@ -176,6 +176,17 @@ public void scalaOutputTest() throws IOException {
     // test code completion
     List<InterpreterCompletion> completions = note.completion(p.getId(), "sc.", 2);
     assertTrue(completions.size() > 0);
+
+    // test cancel
+    p.setText("%spark sc.range(1,10).map(e=>{Thread.sleep(1000); e}).collect()");
+    note.run(p.getId(), false);
+    // wait for 3 seconds so that job is started before cancel
+    Thread.sleep(3000);
```

Comment:
Can you use busy waiting, event or latch here? In the current implementation, this test lasts for six seconds, even if it is not necessary.
Unfortunately, the tests are already slow = (

Reply:
Thanks @aka-demik , I have fixed it